src/core/client/stream/tabs/Comments/Stream/AllCommentsTab/CommentEnteredSubscription.tsx

```diff
@@ -0,0 +1,266 @@
+import { graphql } from "react-relay";
+import {
+  ConnectionHandler,
+  Environment,
+  RecordProxy,
+  RecordSourceSelectorProxy,
+} from "relay-runtime";
+
+import {
+  createSubscription,
+  requestSubscription,
+  SubscriptionVariables,
+} from "coral-framework/lib/relay";
+import { GQLCOMMENT_SORT, GQLCOMMENT_SORT_RL } from "coral-framework/schema";
+
+import { CommentEnteredSubscription } from "coral-stream/__generated__/CommentEnteredSubscription.graphql";
+
+function updateForNewestFirst(
+  store: RecordSourceSelectorProxy<unknown>,
+  storyID: string,
+  storyConnectionKey: string,
+  tag?: string
+) {
+  const rootField = store.getRootField("commentEntered");
+  if (!rootField) {
+    return;
+  }
+  const comment = rootField.getLinkedRecord("comment")!;
+  const commentsEdge = store.create(
+    `edge-${comment.getValue("id")!}`,
+    "CommentsEdge"
+  );
+  commentsEdge.setValue(comment.getValue("createdAt"), "cursor");
+  commentsEdge.setLinkedRecord(comment, "node");
+  const story = store.get(storyID)!;
+  const connection = ConnectionHandler.getConnection(
+    story,
+    storyConnectionKey,
+    {
+      orderBy: GQLCOMMENT_SORT.CREATED_AT_DESC,
+      tag,
+    }
+  )!;
+  const linked = connection.getLinkedRecords("viewNewEdges") || [];
+  connection.setLinkedRecords(linked.concat(commentsEdge), "viewNewEdges");
+}
+
+function updateForOldestFirst(
+  store: RecordSourceSelectorProxy<unknown>,
+  storyID: string,
+  storyConnectionKey: string,
+  tag?: string
+) {
+  const story = store.get(storyID)!;
+  const connection = ConnectionHandler.getConnection(
+    story,
+    storyConnectionKey,
+    {
+      orderBy: GQLCOMMENT_SORT.CREATED_AT_ASC,
+      tag,
+    }
+  )!;
+  const pageInfo = connection.getLinkedRecord("pageInfo")!;
+  // Should not be falsy because Relay uses this information to determine
+  // whether or not new data is available to load.
+  if (!pageInfo.getValue("endCursor")) {
+    // Set cursor to oldest date, to load from the beginning.
+    pageInfo.setValue(new Date(0).toISOString(), "endCursor");
+  }
+  pageInfo.setValue(true, "hasNextPage");
+}
+
+/**
+ * Returns depth until ancestor.
+ */
+function determineDepthTillAncestor(
+  comment: RecordProxy,
+  ancestorID?: string | null
+) {
+  let depth = 0;
+  let cur: RecordProxy | null | undefined = comment;
+  while (cur) {
+    cur = cur.getLinkedRecord("parent");
+    if (cur && cur.getValue("createdAt")) {
```

Comment:
I'm adding this comment
```ts
  // Because we request the parent including its id in the the subscription below
    // we can't just check for truthyness nor check the `id` field of the record to determine
    // that it already exists in our cache. Therefore we check for another field that
    // is part of the `CommentContainer_comment` fragment.
```